### PR TITLE
added dynamic label tests

### PR DESCRIPTION
**Pull request by ChristianKienle**

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/__tests__/UIFont+DynamicTypeTests.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import XCTest
+@testable import ENA
+
+final class UIFont_DynamicTypeTests: XCTestCase {
+    func testWeightFromString() {
+		XCTAssertEqual(UIFont.Weight("ultraLight").rawValue, UIFont.Weight.ultraLight.rawValue, accuracy: .high)
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Extensions/__tests__/UIFont+DynamicTypeTests.swift:25]: Those tests would not be required if `@IBDesignable` + IB would not have been used in the first place. :)

File: src/xcode/ENA/ENA/Source/Views/__tests__/DynamicTypeLabelTests.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		XCTAssertNotNil(DynamicTypeLabel())
+    }
+
+	func testBoldWeight() {
+		let sut = DynamicTypeLabel()
+		sut.dynamicTypeWeight = "bold"
+		// swiftlint:disable:next force_cast
+		let traits = sut.font.fontDescriptor.object(forKey: .traits) as! [UIFontDescriptor.TraitKey: AnyObject]
+		let weight = traits[.weight] as? NSNumber ?? NSNumber(-1)
+		XCTAssertEqual(CGFloat(weight.doubleValue), UIFont.Weight.bold.rawValue, accuracy: 0.001)
```
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Views/__tests__/DynamicTypeLabelTests.swift:34]: Ugly but it works.

File: src/xcode/ENA/ENA/Source/Views/__tests__/DynamicTypeLabelTests.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	func testSemboldWeight() {
```
Comment (SebastianWild) [inline src/xcode/ENA/ENA/Source/Views/__tests__/DynamicTypeLabelTests.swift:37]:
```suggestion
	func testSemiboldWeight() {
```

tiniest of tiny nitpicks :P

**Review by SebastianWild — APPROVED** [review]
Looks good, one spelling nitpick.

**Review by inf2381 — APPROVED** [review]

**Pull request merged**